### PR TITLE
Add arrow navigation to topic detail view

### PR DESCRIPTION
**Pull request by reiterl**
Part of #3447 

Allow in the topic detail view to navigate thru the topics of the meeting.

### CODE REVIEW

**Review by emanuelschuetze — CHANGES_REQUESTED** [review]

File: client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.ts [hunk 1]
```diff
@@ -177,6 +205,45 @@ export class TopicDetailComponent extends BaseMeetingComponent implements OnInit
...
+    }
+
+    public get prevUrl(): any {
+        return `../..`;
+    }
+
+    public getNavDisplay(topic: ViewTopic | null): string | number {
+        if (!!topic) {
+            return !!topic.agenda_item?.item_number ? topic.agenda_item.item_number : topic.sequential_number;
+        }
```
[the remark this answers is not in the document]
Comment (reiterl) [reply client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.ts:243]: Replace the sequential_number with empty string here.
Show only the arrows in the arrow navigation, if there is no number.

File: client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html [hunk 2]
```diff
@@ -128,6 +156,28 @@ <h4>{{ 'Text' | translate }}</h4>
 
 <mat-menu #topicExtraMenu="matMenu">
     <ng-template matMenuContent>
+        <div *ngIf="vp.isMobile && showNavigateButtons">
```
[the remark this answers is not in the document]
Comment (reiterl) [reply client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html:159]: Remove the entries from the menu and always show the arrow navigation in mobile mode.

**Review by Elblinator — CHANGES_REQUESTED** [review]
- The arrow and the number are on not at the same spot. (For the left arrow) The arrow is further left then the number), please display them at the same spot
- If a topic is loaded and then the person is not allowed to see it anymore then the person is still able to navigate through this topic 
(Reproduction: 1. as deli be allowed to mange the agenda -and therefore see hidden topics- 2. open agenda, 3. take away the right to manage the agenda -hidden topic vanishes- navigate through topics -> you can still see the detail view from the hidden topic

File: client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html [hunk 3]
```diff
@@ -19,6 +20,30 @@ <h2>
             </h2>
         </div>
 
+        <!-- Back and forth buttons -->
+        <div *ngIf="!editTopic && showNavigateButtons" class="extra-controls-slot prev-next-topic-controls">
+            <button
+                mat-button
+                (click)="navigateToTopic(previousTopic)"
+                [disabled]="!previousTopic"
+                class="topic-detail-nav-button"
```
Comment (Elblinator) [inline client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html:29]: please remove the unused class

File: client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html [hunk 4]
```diff
@@ -19,6 +20,30 @@ <h2>
...
+            >
+                <os-icon-container icon="chevron_left" [showIcon]="!getNavDisplay(previousTopic)">
+                    {{ getNavDisplay(previousTopic) }}
+                </os-icon-container>
+            </button>
+            <button
+                mat-button
+                (click)="navigateToTopic(nextTopic)"
+                [disabled]="!nextTopic"
+                class="topic-detail-nav-button"
```
Comment (Elblinator) [inline client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html:39]: please remove the unused class

**Review by Elblinator — CHANGES_REQUESTED** [review]

File: client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html [hunk 5]
```diff
@@ -19,6 +20,30 @@ <h2>
             </h2>
         </div>
 
+        <!-- Back and forth buttons -->
+        <div *ngIf="!editTopic && showNavigateButtons" class="extra-controls-slot prev-next-topic-controls">
```
Comment (Elblinator) [inline client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html:24]: additional unused class

**Comment by reiterl** [comment]
I need https://github.com/OpenSlides/openslides-backend/issues/2352 for a setting as switch for the agenda item arrow navigation.

File: client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html (hunk 3 above)
Comment (reiterl) [reply client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html]: done

File: client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html (hunk 4 above)
Comment (reiterl) [reply client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html]: done

File: client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html (hunk 5 above)
Comment (reiterl) [reply client/src/app/site/pages/meetings/pages/agenda/modules/topics/pages/topic-detail/components/topic-detail/topic-detail.component.html]: done

**Comment by reiterl** [comment]
I updated the sorting of the navigation. I talked with @luisa-beerboom about the problem with deleted agenda items in the `ViewModelListObservable`. It could be a problem with the repository. @bastianjoel  let us talk about it.

**Comment by reiterl** [comment]
Also I need to introduce the setting logic.

**Comment by reiterl** [comment]
I have update the services and now the delete topic/still visible is no longer an issue. I cannot reproduce it any longer.
I have talked with @bastianjoel  about is. It needs a test.

Open is the missing setting stuff.

**Comment by reiterl** [comment]
The show/hide setting  part of this issue will be done in a different issue. Else this feature is complete.

**Review by Elblinator — CHANGES_REQUESTED** [review]
this is seems to be an unrelated issue <s>a delegate is always able to see hidden topics, hidden topics should only be visible if the participant is allowed to manage the agenda</s>

the arrows are not visible at all if: 
1. navigate to topic view
2. shrink window to mobile view 
3. reload page
4. arrows are not visible
5. arrows stay invisible even if the view is widened

**Comment by Elblinator** [comment]
maybe look at motion-detail-view.component for a reference for the arrows

**Review by bastianjoel — APPROVED** [review]

**Pull request merged**